libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFJsonUtils.h

```diff
@@ -67,7 +67,16 @@
 
 
 /**
- Pull a value from the json-derived object by path ("." delimited)
+ * Pull a value from the json-derived object by path ("." delimited)
+ *
+ * Examples (in pseudo code):
+ *
+ * json = {"a": {"b": [{"c":"xx"}, {"c":"xy"}, {"d": [{"e":1}, {"e":2}]}, {"d": [{"e":3}, {"e":4}]}] }}
+ * projectIntoJson(jsonObj, "a") = {"b": [{"c":"xx"}, {"c":"xy"}, {"d": [{"e":1}, {"e":2}]}, {"d": [{"e":3}, {"e":4}]} ]}
+ * projectIntoJson(json, "a/b") = [{c:"xx"}, {c:"xy"}, {"d": [{"e":1}, {"e":2}]}, {"d": [{"e":3}, {"e":4}]}]
+ * projectIntoJson(json, "a/b/c") = ["xx", "xy"]                                     // new in 4.1
```

Comment:
I'm not sure what the syntax would be.

Basically the change I'm making is to say that:
project (arr, path) -> [project(elt,path) for elt in arr]
Up to now, all we had was something like:
project(dict, (headPath | tailPath) -> project(dict[headPath], tailPath)